pulsar-proxy/src/test/java/org/apache/pulsar/proxy/server/ProxyTest.java

```diff
@@ -259,6 +259,34 @@ public void testRegexSubscription() throws Exception {
         }
     }
 
+    @Test(timeOut = 60_000)
+    public void testRegexSubscriptionWithTopicDiscovery() throws Exception {
+        @Cleanup
+        PulsarClient client = PulsarClient.builder().serviceUrl(proxyService.getServiceUrl()).build();
+        String subName = "regex-proxy-test-" + System.currentTimeMillis();
+        String regexSubscriptionPattern = "persistent://sample/test/local/.*";
+        try (Consumer<byte[]> consumer = client.newConsumer()
+                .topicsPattern(regexSubscriptionPattern)
+                .subscriptionName(subName)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .patternAutoDiscoveryPeriod(10, TimeUnit.MINUTES)
```

Comment:
Interesting, it looks like we don't have a way to disable the auto-discovery task. We should have a way to disable it since we have topic list watcher after 3.0.

Reply:
yes, I will discuss supporting it in the ML after this pr get merged.